tools/generator/filters/java.py

```diff
@@ -0,0 +1,131 @@
+# SPDX-License-Identifier: GPL-2.0-or-later
+# Copyright (c) 2020 Association Prologin <association@prologin.org>
+# Copyright (c) 2020 Sacha Delanoue
+# Copyright (c) 2020 Antoine Pietri
+
+from functools import partial
+
+from . import register_filter
+from .common import camel_case, generic_args, generic_prototype, is_array
+from .cxx import cxx_comment, cxx_type
+
+
+@register_filter
+def java_type(value: str) -> str:
+    if value in ('byte', 'char', 'short', 'int', 'long', 'float', 'double', 'void'):
+        return value
+    if value == 'bool':
+        return "boolean"
+    if value == "string":
+        return "String"
+    if value.endswith(" array"):
+        return "{}[]".format(java_type(value[:-len(' array')]))
+    return camel_case(value)
+
+
+@register_filter
+def java_args(value, type_mapper=lambda x: x) -> str:
+    return generic_args(value, type_mapper=java_type)
+
+
+java_prototype = register_filter(
+    partial(generic_prototype, type_mapper=java_type, arg_mapper=java_args),
+    name='java_prototype',
+)
+
+java_comment = register_filter(cxx_comment, name='java_comment')
+
+
+def _jtype(typename: str) -> str:
+    """Return the type to be used as Lang in lang2cxx and cxx2lang"""
+    generics = {
+        'bool': 'jboolean',
+        'double': 'jdouble',
+        'int': 'jint',
+        'string': 'jstring',
+    }
+    if typename in generics.keys():
+        return generics[typename]
+    return camel_case(typename)
+
+
+def _cxx_type(value: str) -> str:
```

Comment:
Could you explain the need for this?

Reply:
done

Reply:
Two reasons why it is done at this level:
The first one is that it is how the ruby generator does it.
The second, is that, because of lines like `jrt.env->SetBooleanArrayRegion(out, (jsize)0, (jsize)in.size(), (const jboolean*)in.data());`, if `in` is a `std::vector<bool>`, then `.data()` is not a `bool*` and then the entire behavior changes.

So is it possible to change the behavior so that we see only `bool` and not `boolean` in the C++? Yes. Is it fast and simple? No.

Reply:
Ok I removed the whole `boolean` stuff. Now it uses `bool` for C++, and it works.